### PR TITLE
GH-3458: Remove ExtendedVariable class

### PR DESCRIPTION
**Pull request by fkleedorfer**
* all occurrences replaced with Sparqlbuilder's Variable class
* changed Variable class:
    - public constructor
    - getter for field varName
    - changed field name from alias to varName (closer to spec)

GitHub issue resolved: #3458 
Briefly describe the changes proposed in this PR:

`ExtendedVariable` was introduced by the rdf4j-spring code to access the constructor and `varName` property. Making these things accessible to the outside directly in the `Variable` class seems cleaner.

----
PR Author Checklist (see the [contributor guidelines](https://github.com/eclipse/rdf4j/blob/main/.github/CONTRIBUTING.md) for more details):

 - [x] my pull request is [self-contained](https://rdf4j.org/documentation/developer/merge-strategy/#self-contained-changes-pull-requests-and-commits)
 - [x] I've added tests for the changes I made
 - [x] I've applied [code formatting](https://github.com/eclipse/rdf4j/blob/main/.github/CONTRIBUTING.md#code-formatting) (you can use `mvn process-resources` to format from the command line)
 - [x] I've [squashed](https://rdf4j.org/documentation/developer/squashing) my commits where necessary 
 - [x] every commit message starts with the issue number (GH-xxxx) followed by a meaningful description of the change

### CODE REVIEW

**Comment by fkleedorfer** [comment]
In this PR, the Constructor of the `Variable` class is made public. Maybe that's a bit much. Alternatively we could enforce creation of variables through `SparqlBuilder.var()`. However, for the `rdf4j-spring` package, we need the accessor for the `varName` field and some way of instantiating variables.

**Review by jeenbroekstra — APPROVED** [review]

File: core/sparqlbuilder/src/main/java/org/eclipse/rdf4j/sparqlbuilder/core/Variable.java [hunk 1]
```diff
@@ -21,15 +21,19 @@
  */
 public class Variable implements Projectable, RdfSubject, RdfPredicate, RdfObject, Operand, Orderable, Groupable,
 		GraphName, Assignable {
-	private String alias;
+	private String varName;
 
-	Variable(String varName) {
-		this.alias = varName;
+	public Variable(String varName) {
```
Comment (jeenbroekstra) [inline core/sparqlbuilder/src/main/java/org/eclipse/rdf4j/sparqlbuilder/core/Variable.java:26]: I have no _very_ strong feelings on this, but if you can leave it package-private without having to jump through too many hoops, then I think I'd prefer sticking with that.

**Review by jeenbroekstra — COMMENTED** [review]
LGTM. As soon as we decide on the constructor visibility one way or the other this is good to go I think.

File: core/sparqlbuilder/src/main/java/org/eclipse/rdf4j/sparqlbuilder/core/Variable.java (hunk 1 above)
Comment (fkleedorfer) [reply core/sparqlbuilder/src/main/java/org/eclipse/rdf4j/sparqlbuilder/core/Variable.java:26]: Ok, I'll make it package-private (thanks for making me look that up btw :-)) and we'll go through `SparqlBuilder.var()` in `rdf4j-spring`.

Comment (fkleedorfer) [reply core/sparqlbuilder/src/main/java/org/eclipse/rdf4j/sparqlbuilder/core/Variable.java]: done

**Pull request merged**